src/test/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSparkUnitTest.java

```diff
@@ -0,0 +1,102 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import htsjdk.samtools.metrics.MetricsFile;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.broadinstitute.hellbender.tools.picard.analysis.InsertSizeMetrics;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+public final class CollectInsertSizeMetricsSparkUnitTest extends CommandLineProgramTest{
+
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "picard/analysis/CollectInsertSizeMetrics");
+
+    public String getTestedClassName() {
+        return CollectInsertSizeMetricsSpark.class.getSimpleName();
+    }
+
+    @DataProvider(name="metricsfiles")
+    public Object[][] insertSizeMetricsFiles() {
+        return new Object[][] {
+                // {"insert_size_metrics_test.sam", null},
```

Comment:
They are not used:
1. sam is the same as bam (remember the empty bam is regenerated from sam?) so testing it is equivalent to testing other parts of GATK
2. the case with reference is not yet implemented or checked--but planned to be.